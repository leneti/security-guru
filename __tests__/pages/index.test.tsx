import { render, screen, fireEvent } from "@testing-library/react";
import HomePage from "@site/pages";

jest.mock("@site/components/Hero");
jest.mock("@site/components/PageTitle");
jest.mock("@site/components/Services");

describe("Home page", () => {
  it.each([
    { state: "with animation", value: true },
    { state: "without animation", value: false },
  ])("renders scroll button $state", ({ value }) => {
    jest
      .spyOn(jest.requireActual("@mantine/hooks"), "useReducedMotion")
      .mockReturnValue(value);

    const { container } = render(<HomePage />);
    expect(container).toMatchSnapshot();
  });

  it("scrolls to services on click", () => {
    const mockScrollFn = jest.fn().mockName("scrollIntoView");
    jest
      .spyOn(jest.requireActual("@mantine/hooks"), "useScrollIntoView")
      .mockReturnValue({ scrollIntoView: mockScrollFn });

    render(<HomePage />);

    const scrollBtn = screen.getByTitle(/eiti Å¾emyn/i);
    fireEvent.click(scrollBtn);
    expect(mockScrollFn).toHaveBeenCalled();
  });
});
