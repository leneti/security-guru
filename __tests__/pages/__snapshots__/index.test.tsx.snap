// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders scroll button with animation 1`] = `
<div>
  <title>
    Mock SG-Test
  </title>
  <div>
    Mock Hero
  </div>
  <button
    class="mantine-UnstyledButton-root mantine-956te5"
    title="Eiti žemyn"
    type="button"
  />
  <div
    class="mantine-Container-root mantine-1hy4it3"
  >
    <h2
      class="mantine-Text-root mantine-Title-root mantine-44lm1w"
    >
      Paslaugos
    </h2>
    <div
      class="mantine-Divider-root mantine-Divider-horizontal mantine-1ah7ro1"
      role="separator"
    />
    <div>
      Mock Services
    </div>
  </div>
  <div
    class="mantine-Center-root mantine-1y5prk2"
  >
    <a
      class="mantine-UnstyledButton-root mantine-Button-root mantine-bouam0"
      data-button="true"
      href="/susisiekite"
      type="button"
    >
      <div
        class="mantine-1wpc1xj mantine-Button-inner"
      >
        <span
          class="mantine-1ryt1ht mantine-Button-label"
        >
          Susisiekite
        </span>
        <span
          class="mantine-Button-icon mantine-Button-rightIcon mantine-467wo8"
        >
          <svg
            class="tabler-icon tabler-icon-chevron-right"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
        </span>
      </div>
    </a>
  </div>
</div>
`;

exports[`Home page renders scroll button without animation 1`] = `
<div>
  <title>
    Mock SG-Test
  </title>
  <div>
    Mock Hero
  </div>
  <button
    class="mantine-UnstyledButton-root mantine-exdozc"
    title="Eiti žemyn"
    type="button"
  />
  <div
    class="mantine-Container-root mantine-1hy4it3"
  >
    <h2
      class="mantine-Text-root mantine-Title-root mantine-44lm1w"
    >
      Paslaugos
    </h2>
    <div
      class="mantine-Divider-root mantine-Divider-horizontal mantine-1ah7ro1"
      role="separator"
    />
    <div>
      Mock Services
    </div>
  </div>
  <div
    class="mantine-Center-root mantine-1y5prk2"
  >
    <a
      class="mantine-UnstyledButton-root mantine-Button-root mantine-bouam0"
      data-button="true"
      href="/susisiekite"
      type="button"
    >
      <div
        class="mantine-1wpc1xj mantine-Button-inner"
      >
        <span
          class="mantine-1ryt1ht mantine-Button-label"
        >
          Susisiekite
        </span>
        <span
          class="mantine-Button-icon mantine-Button-rightIcon mantine-467wo8"
        >
          <svg
            class="tabler-icon tabler-icon-chevron-right"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
        </span>
      </div>
    </a>
  </div>
</div>
`;
